sim/pokemon.ts

```diff
@@ -1356,6 +1360,10 @@ export class Pokemon {
 				this.ability = ''; // Don't allow Illusion to wear off
 			}
 			this.setAbility(species.abilities['0'], null, true);
+			this.preFormeChangeInfo = {
+				species: preFormeChangeSpecies,
+				ability: this.baseAbility,
```

Comment:
How does these differ from `this.set.species` and `this.set.ability`?

Reply:
the Species here is a Species class, not much difference but I think I prefer it so it doesn't need to be grabbed with species.get again, there's no real difference for ability but again I like using the existing properties in the class if/when possible